lang/java/reef-io/src/main/java/org/apache/reef/io/network/naming/NameServerImpl.java

```diff
@@ -75,8 +79,10 @@ private NameServerImpl(
     final Codec<NamingMessage> codec = NamingCodecFactory.createFullCodec(factory);
     final EventHandler<NamingMessage> handler = createEventHandler(codec);
 
-    injector.bindVolatileParameter(RemoteConfiguration.HostAddress.class, localAddressProvider.getLocalAddress());
+    String host = UNKNOWN_HOST_NAME.equals(hostAddress) ? localAddressProvider.getLocalAddress() : hostAddress;
+    injector.bindVolatileParameter(RemoteConfiguration.HostAddress.class, host);
```

Comment:
Please explain why this is necessary.

Reply:
When creating the object of NettyMessagingTransport in line 90, we need the HostAddress to refer to the hostname at all times (irrespective of whether this is running within a container or not) since we create a server endpoint and bind to it. Otherwise, the bind will fail.

Since we have two ways to get the hostname (using HostAddress as well as LocalAddressProvider), I bound to HostAddress in AzureBatchBootstrapREEFLauncher class and used the value here.

In other cases, the HostAddress is not bound - so it is UNKNOWN which will read from localAddressProvider.getLocalAddress(). So the existing implementation is not impacted due to this change.

---
In reply to: [201087914](https://github.com/sharathmalladi/reef/pull/5#discussion_r201087914) [](ancestors = 201087914)